apps/voting/app/src/math-utils.js

```diff
@@ -32,13 +33,103 @@ export function formatNumber(num, decimals = 2, { truncate = true } = {}) {
       }`
     : whole
 
-  // If we were dealing with a yuge number, append the exponent suffix back
+  // If we were dealing with a huge number, append the exponent suffix back
   return exponentialIndex > -1
     ? `${formattedNumber}${numString.substring(exponentialIndex)}`
     : formattedNumber
 }
+export class Percentage {
+  /**
+   * A value represented as Percentage using a specific base unit
+   * @param {string} value source value
+   * @param {string} base base unit
+   */
+  constructor(value, base) {
+    this._value = new BN(value)
+    this._base = new BN(base)
+  }
+
+  base() {
+    return this._base
+  }
+
+  value() {
+    return this._value
+  }
+
+  valueOf() {
+    return this.toNumber()
+  }
+
+  /**
+   * Parses represented percentage value to string
+   *
+   * @param {{ digits: number, suffix: string }} formattingOptions digit and suffix to be use in formatting
+   * @returns {string} Formatted value
+   */
+  toString({ digits = 2, suffix = '%' } = {}) {
+    const MAX_BASE_PRECISION = 10 ** 15 // max value supported for a base on Big Number libraries
+    let basePrecision = 10 ** digits
 
-export function percentageList(values, digits = 0) {
+    // Tolerate having too many digits by correcting the value.
+    if (basePrecision > Number.MAX_SAFE_INTEGER) {
+      basePrecision = MAX_BASE_PRECISION
+    }
+
+    return (
+      formatNumber(
+        new Percentage(this._value.mul(new BN(100)), this._base).toNumber(
+          basePrecision
+        ),
+        digits
+      ) + suffix
+    )
+  }
+
+  /**
+   * Converts a percentage into a number between 0 and 1.
+   * @param {number} precision precision to use for the parsing
+   * @returns {number} parsed value
+   */
+  toNumber(precision = 10 ** 9) {
+    return (
+      parseInt(
+        divideRoundBigInt(this._value.mul(new BN(precision)), this._base),
+        10
+      ) / precision
+    )
+  }
+}
+
```

Comment:
should there be func docs here for TokenAmount?

Reply:
Yes it should, I will be adding one

Reply:
Done, can you check?